Develop

IvanTcanga opened the pull request.

IvanTcanga committed:
Актуальные изменения в develop для сдачи спринта

IvanTcanga committed:
Update README.md

Добавление README

AlexKitIt commented on tests/test_order_page.py:
```diff
@@ -0,0 +1,24 @@
+    def test_order_all_fields_success(self, driver, button, test_user):
+        order_page = OrderPage(driver)
+        order_page.scroll_to_element(button)
+        order_page.click_to_element(OrderPageLocators.CONFIRM_COOCKIE)
```
Нужно исправить: здесь и далее: методов с принимаемыми локаторами в тестах быть не должно, если это не вариант параметризации, они живут в методах пейджи. В тесте только создание объекта и вызов метода, без указания локаторов элементов с которым происходит взаимодействие. Тест должен быть максимально изолирован от логики выполнения

IvanTcanga replied:
Спасибо за совет,исправил.

AlexKitIt commented on tests/test_redirect_pages.py:
```diff
@@ -0,0 +1,24 @@
+        main_page.find_element_with_wait(MainPageLocators.LOGO_YANDEX)
+        main_page.click_to_element(MainPageLocators.LOGO_YANDEX)
+        main_page.switch_to_next_tab()
+        assert 'dzen' in driver.current_url
```
Нужно исправить: здесь и далее: прямого обращения к driver, вызова его property, execute_script не нем и тд из тестов быть не должно. Для этого в base пишутся соответствующие методы и тут вызываются на объектах конкретных страниц

IvanTcanga replied:
Исправил: добавил новые методы,работу с property driver убрал в base_page.

IvanTcanga committed:
Update test_order_page.py

Исправлены замечания касательно локаторов в тестах,за исключением параметризации, удалена лишняя логика.

IvanTcanga committed:
Update test_redirect_pages.py

Исправлены замечания по поводу локаторов в тестах.

IvanTcanga committed:
Update base_page.py

Добавлен метод для получения текущего урла.

IvanTcanga committed:
Update main_page.py

Добавлены методы для устранения замечаний по локаторам и driver в тестах.
Оптимизация логики.

IvanTcanga committed:
Update main_page_locators.py

IvanTcanga committed:
Update main_page.py

IvanTcanga committed:
Update test_order_page.py

Добавлен куки кликер

IvanTcanga committed:
Update test_order_page.py

IvanTcanga committed:
Update main_page.py

IvanTcanga committed:
Update main_page_locators.py

IvanTcanga committed:
Update order_page.py

AlexKitIt commented on pages/order_page.py:
```diff
@@ -0,0 +1,57 @@
+        self.find_element_with_wait(OrderPageLocators.BUTTON_CONFIRM)
+        self.click_to_element(OrderPageLocators.BUTTON_CONFIRM)
+
+    @allure.step('Принимаем куки')
+    def coockie_clicker(self):
+        return self.click_to_element(MainPageLocators.CONFIRM_COOCKIE)
```
Нужно исправить:  в классе описываются только методы работы с элементами на конкретной пейдже, методов для работы с элементами с других страниц быть не должно, иными словами в OrderPage не должно быль методов для работы с локаторами с MainPage
